packages/nuxt/src/app/composables/payload.ts

```diff
@@ -19,35 +19,43 @@ interface LoadPayloadOptions {
 /** @since 3.0.0 */
 export async function loadPayload (url: string, opts: LoadPayloadOptions = {}): Promise<Record<string, any> | null> {
   if (import.meta.server || !payloadExtraction) { return null }
-  const payloadURL = await _getPayloadURL(url, opts)
-  const nuxtApp = useNuxtApp()
-  const cache = nuxtApp._payloadCache ||= {}
-  if (payloadURL in cache) {
-    return cache[payloadURL] || null
+  // TODO: allow payload extraction for non-prerendered URLs
+  const shouldLoadPayload = await isPrerendered(url)
+  if (!shouldLoadPayload) {
+    return null
   }
-  cache[payloadURL] = isPrerendered(url).then((prerendered) => {
-    if (!prerendered) {
-      cache[payloadURL] = null
-      return null
-    }
-    return _importPayload(payloadURL).then((payload) => {
-      if (payload) { return payload }
-
-      delete cache[payloadURL]
-      return null
-    })
-  })
-  return cache[payloadURL]
+  const payloadURL = await _getPayloadURL(url, opts)
+  return await _importPayload(payloadURL) || null
+}
+let linkRelType: string | undefined
+function detectLinkRelType () {
```

Comment:
can this be run server side too ?

Reply:
yes. preloadPayload has server handling

Reply:
oh my bad, didn't see the `import.meta.server` flag